### PR TITLE
Switch from Months to Years at 2 years old

### PR DESCRIPTION
**Pull request by AronNovak**

### CODE REVIEW

File: client/src/elm/Pages/WellChildProgressReport/View.elm [hunk 1]
```diff
@@ -617,10 +617,7 @@ resolveDateOfLastNutritionAssessment currentDate isChw initiator mandatoryNutrit
 generatePartitionedWarningEntries :
     ModelIndexedDb
     -> Maybe AssembledData
-    ->
-        ( List ( NominalDate, PediatricCareMilestone, EncounterWarning )
-        , List ( NominalDate, PediatricCareMilestone, EncounterWarning )
-        )
+    -> ( List ( NominalDate, PediatricCareMilestone, EncounterWarning ), List ( NominalDate, PediatricCareMilestone, EncounterWarning ) )
```
Comment (anvmn) [inline client/src/elm/Pages/WellChildProgressReport/View.elm:620]: Why this was changed?

File: client/src/elm/Utils/NominalDate.elm [hunk 2]
```diff
@@ -94,6 +94,34 @@ renderAgeMonthsDaysParts language birthDate now =
     [ monthPart, dayPart ]
 
 
+renderAgeYearsMonths : Language -> NominalDate -> NominalDate -> String
+renderAgeYearsMonths language birthDate now =
```
Comment (anvmn) [inline client/src/elm/Utils/NominalDate.elm:98]: This is a generic function and it should be covering all options.
It doesn't matter that we're using it only from 2 years.

Options are:
0 / 1 /  1+ years
0 / 1 /  1+ months

Should be something like: 

```
case years of 
  0 ->
    case months of
        0 ->
          ..

        1 ->
          ..
        
        _ ->
          ..
          

  1 ->
      ..


  _ ->
      ..
```

**Review by anvmn — CHANGES_REQUESTED** [review]

**Review by anvmn — CHANGES_REQUESTED** [review]
@Diama1 

There's one other place where this needs to be changed:

Peek 2021-11-28 10-45

File: client/src/elm/Pages/WellChildProgressReport/View.elm (hunk 1 above)
Comment (Diama1) [reply client/src/elm/Pages/WellChildProgressReport/View.elm:620]: > 

@anvmn just on save elm-format is changing it

Comment (anvmn) [reply client/src/elm/Pages/WellChildProgressReport/View.elm:620]: @Diama1 which version are you using?

Comment (Diama1) [reply client/src/elm/Pages/WellChildProgressReport/View.elm:620]: @anvmn Elm-format 0.8.3

Comment (anvmn) [reply client/src/elm/Pages/WellChildProgressReport/View.elm:620]: @Diama1 please update to `0.8.5`

File: client/src/elm/Utils/NominalDate.elm [hunk 3]
```diff
@@ -94,6 +94,42 @@ renderAgeMonthsDaysParts language birthDate now =
...
+        0 ->
+            case months of
+                1 ->
+                    translate language <| Translate.AgeSingleMonthWithoutDay months
+
+                _ ->
+                    translate language <| Translate.AgeMonthsWithoutDay months
+
+        1 ->
+            translate language <| Translate.AgeMonthsWithoutDay months
```
Comment (anvmn) [inline client/src/elm/Utils/NominalDate.elm]: This is wrong

**Review by anvmn — CHANGES_REQUESTED** [review]

File: client/src/elm/Utils/NominalDate.elm [hunk 4]
```diff
@@ -94,6 +94,47 @@ renderAgeMonthsDaysParts language birthDate now =
...
+    case years of
+        0 ->
+            case months of
+                1 ->
+                    translate language <| Translate.AgeSingleMonthWithoutDay months
+
+                _ ->
+                    translate language <| Translate.AgeMonthsWithoutDay months
+
+        1 ->
```
Comment (anvmn) [inline client/src/elm/Utils/NominalDate.elm:118]: @Diama1 

Still wrong. It should be saying one of:

- One year old
- One year and one month old
- One year and X months old

**Review by anvmn — CHANGES_REQUESTED** [review]

File: client/src/elm/Translate.elm [hunk 5]
```diff
@@ -331,6 +331,9 @@ type TranslationId
     | AgeSingleMonthWithoutDay Int
     | AgeSingleDayWithMonth Int Int
     | AgeSingleDayWithoutMonth Int Int
+    | AgeSingleYearOld Int
```
Comment (anvmn) [inline client/src/elm/Translate.elm]: Why we need `Int`?
It's a single year, so it's translates to `One year`.

File: client/src/elm/Translate.elm [hunk 6]
```diff
@@ -331,6 +331,9 @@ type TranslationId
...
+    | AgeSingleYearWithSingleMonth Int Int
```
Comment (anvmn) [inline client/src/elm/Translate.elm]: Why we need 2 `Int`?
It's a single year, so it translates to `One year and one month`.

File: client/src/elm/Translate.elm [hunk 7]
```diff
@@ -331,6 +331,9 @@ type TranslationId
...
+    | AgeSingleYearWithMonths Int Int
```
Comment (anvmn) [inline client/src/elm/Translate.elm]: Why we need first `Int`?
It's a single year, so it's translates to `One year and ...`.

**Review by anvmn — CHANGES_REQUESTED** [review]
@Diama1 

Less copy/paste please

File: client/src/elm/Translate.elm [hunk 8]
```diff
@@ -2007,19 +2007,19 @@ translationSet trans =
...
+        AgeOneYearAndOneMonth ->
+            { english = "One year and one month"
+            , kinyarwanda = Just " Umwaka n'ukwezi kumwe"
             }
 
-        AgeSingleYearWithMonths year months ->
-            { english = String.fromInt year ++ " years " ++ String.fromInt months ++ " months"
-            , kinyarwanda = Just <| " Umwaka " ++ String.fromInt year ++ " Amezi " ++ String.fromInt months
+        AgeOneYearWithMonths months ->
+            { english = "1 year " ++ String.fromInt months ++ " months"
```
Comment (anvmn) [inline client/src/elm/Translate.elm]: => `One year and ...`

File: client/src/elm/Translate.elm [hunk 9]
```diff
@@ -2007,19 +2007,19 @@ translationSet trans =
...
+            , kinyarwanda = Just <| " Umwaka n'amezi " ++ String.fromInt months
```
Comment (anvmn) [inline client/src/elm/Translate.elm]: Drop space at `" Umwaka`

File: client/src/elm/Translate.elm [hunk 10]
```diff
@@ -2007,19 +2007,19 @@ translationSet trans =
...
+            { english = "One year old"
+            , kinyarwanda = Just "Umwaka umwe"
             }
 
-        AgeSingleYearWithSingleMonth year month ->
-            { english = String.fromInt year ++ " year " ++ String.fromInt month ++ " month"
-            , kinyarwanda = Just <| " Umwaka " ++ String.fromInt year ++ " Ukwezi " ++ String.fromInt month
+        AgeOneYearAndOneMonth ->
+            { english = "One year and one month"
+            , kinyarwanda = Just " Umwaka n'ukwezi kumwe"
```
Comment (anvmn) [inline client/src/elm/Translate.elm]: Drop space at `" Umwaka`

**Review by anvmn — CHANGES_REQUESTED** [review]

**Comment by anvmn** [comment]
@Diama1 Hopefully ^^ is the last set of fixes.
Also, please resolve conflicts.

**Review by anvmn — APPROVED** [review]
Thanks @Diama1

**Pull request merged**